be/src/storage/tablet_updates.cpp

```diff
@@ -1516,6 +1516,7 @@ Status TabletUpdates::compaction(MemTracker* mem_tracker) {
     if (!_compaction_running.compare_exchange_strong(was_runing, true)) {
         return Status::InternalError("illegal state: another compaction is running");
     }
+    std::unique_lock lock(get_update_compaction_lock(), std::try_to_lock);
```

Comment:
Why do you call try_to_lock. What will happen if the lock is not held successfully?

Reply:
A bug does exist here. I submit a new pr, if the lock is not held successfully, the update compaction will quit.